card/lib/card/format/context_names.rb

```diff
@@ -1,23 +1,25 @@
 class Card
   class Format
     module ContextNames
-
       def context_names
         @context_names ||= initial_context_names
       end
 
       def initial_context_names
-        @initial_context_names ||=
-          parent ? context_names_from_parent : context_names_from_params
+        @initial_context_names ||= relevant_context_names do
+          parent ? parent.context_names : context_names_from_params
+        end
       end
 
-      def context_names_from_parent
+      def relevant_context_names
```

Comment:
this is an important step towards making it so that context names are handled entirely with the voo, because it gets rid of names that would needlessly kill caching.

having context names in the voo should eliminate some caching bugs where we see different variants of names than what we expect.